File: 7-kyu/correct-the-time-string/swift/solution.swift
import Foundation

func correct(_ timeString: String?) -> String? {
  guard let timeString = timeString else { return nil }
  if timeString.isEmpty {
    return ""
  }
  let timeSplit = timeString.components(separatedBy: ":").compactMap { Int($0) }
  if timeSplit.count != 3 {
    return nil
  }
  let totalSeconds = timeSplit[0] * 3600 + timeSplit[1] * 60 + timeSplit[2]
  let hours = totalSeconds / 3600 % 24
  let minutes = totalSeconds / 60 % 60
  let seconds = totalSeconds % 60
  return String(format: "%02d:%02d:%02d", hours, minutes, seconds)
}

File: 7-kyu/correct-the-time-string/python/solution.py
import re

def time_correct(t):
    if not t:
        return None if t is None else ""
    elif re.fullmatch(r"[0-9]{2}:[0-9]{2}:[0-9]{2}", t) is None:
        return None
    hours, minutes, seconds = [int(n) for n in t.split(':')]
    if hours > 23 or minutes > 59 or seconds > 59:
        minutes += seconds // 60
        seconds %= 60
        hours += minutes // 60
        minutes %= 60
        hours %= 24
    return f'{hours:02}:{minutes:02}:{seconds:02}'
